src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import weakref
+from collections import defaultdict
+from queue import Empty, Queue
+from threading import Event, Thread
+from typing import TYPE_CHECKING, Dict, Optional
+
+from snowflake.snowpark._internal.analyzer.snowflake_plan_node import SnowflakeTable
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session  # pragma: no cover
+
+
+class TempTableAutoCleaner:
+    """
+    Automatically cleans up unused temporary tables created in the current session
+    when the DataFrame is no longer referenced (i.e., gets garbage collected).
```

Comment:
DataFrame -> temp table, this component doesn't use the dataframe concept at all.

Reply:
done